### PR TITLE
Ecdc 3446 new filewriter attribute

### PR DESCRIPTION
**Pull request by ggoneiESS**
### Issue

Close ECDC-3446*

### Description of work

Added Filewriter option in module dropdown and associated logic for validation, tree, JSON, and tests.  As part of this, small changes to UI stylesheets and Python environment packages.

### Acceptance Criteria 
Filewriter should appear ultimately in JSON, and be clearly editable and visible to user as being present.  Attached to the Filewriter option is a simple name.

### UI tests

Small change to match new text colour for validations (so it is visible to users who are utilising dark mode)

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: nexus_constructor/model/module.py [hunk 1]
```diff
@@ -167,6 +168,24 @@ class F144Stream(F142Stream):
...
+
+    def as_dict(self, error_collector: List[str]):
+        return {
+            CommonKeys.MODULE: self.writer_module,
+            NodeType.CONFIG: {CommonKeys.NAME: self.name},
+        }
+
+    def as_nexus(self, nexus_node, error_collector: List[str]):
+        nexus_dataset = nexus_node.create_dataset(self.name, data=self.name)
+        nexus_dataset.attrs[self.name] = self.name
```
[the remark this answers is not in the document]
Comment (ggoneiESS) [reply nexus_constructor/model/module.py:186]: I think this will be removed, yes, but it was needed for now as program doesn't behave well if the class is missing this

File: tests/test_validators.py [hunk 2]
```diff
@@ -173,6 +173,21 @@ def test_GIVEN_valid_string_value_WHEN_validating_field_value_THEN_returns_accep
...
+        field_type_combo, dataset_type_combo, FieldType.filewriter.value
+    )
+    validator.is_valid = Mock()
+
+    assert validator.validate(strvalue, 0) == QValidator.Acceptable
+    validator.is_valid.emit.assert_called_once_with(True)
+
+
 def test_GIVEN_invalid_float_value_WHEN_validating_field_value_THEN_returns_intermediate_and_emits_signal_with_false():
     invalid_value = "sdfn"
```
[the remark this answers is not in the document]
Comment (ggoneiESS) [reply tests/test_validators.py:192]: Agreed, I think one of the things I will do during the restructure is to ensure better test coverage.  UI checks failed on macOS due to the adjustment for dark mode, will chat with Afonso when he's here about how exactly the environments are set but in principle the OS should have some flag that Qt accesses to check for dark mode

File: nexus_constructor/model/module.py [hunk 3]
```diff
@@ -167,6 +168,24 @@ class F144Stream(F142Stream):
     writer_module = attr.ib(type=str, default=WriterModules.F144.value, init=False)
 
 
+@attr.s
+class FileWriter(FileWriterModule):
+    name = attr.ib(type=str)
+    type = attr.ib(type=str, default="string")
+    values = attr.ib(type=str, default=None)
```
[the remark this answers is not in the document]
Comment (ggoneiESS) [reply nexus_constructor/model/module.py:175]: type, values are needed for the add component workflow to pass successfully without throwing a TypeError (it is initialised needing these) and so they need to be initialised in this class and added as attributes.  Not ideal, could in principle be set to static values of "string" and 'None', but this allows a bit of flexibility.  I would suggest leaving it here for now and keeping it in mind for a later ticket when the classes are restructured.

File: tests/test_validators.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ggoneiESS) [reply tests/test_validators.py:192]: I think it might be worthwhile making another ticket for test coverage, as currently none of the field types are checked (and other issues exist too).  I've added it to a JSON string though to make sure it doesn't at least cause issues on import.
